### DISCUSSION TITLE
Supabase Edge functions - error "relation does not exist"

### DISCUSSION THREAD

**Discussion by ErwanJes** [supabase/supabase#15812] [Questions]
I'm trying to query my database from an Edge function without a JWT.
I'm not sure why but I cannot trigger any kind of a request (select, update...) in that function.

For example, if I call this:
``` 
  const supabase = createClient(
    Deno.env.get("SUPABASE_URL") ?? "",
    Deno.env.get("SUPABASE_SERVICE_ROLE_KEY") ?? "",
  );
  const { data, error } = await supabase.from('profiles').select('*')
```
`profiles` is a table in `public` filled with multiples rows but it returns null.
The error is
```
  error: {
    code: "42P01",
    details: null,
    hint: null,
    message: 'relation "public.profiles" does not exist'
  }
  ```

I guess it comes from the fact there is not user's token but I'm not sure.
The function is called from a webhook so I don't have a JWT that I can use.

Can anyone confirm it's linked to a missing JWT? If so how can I bypass that?
Or did I forget something else?

I don't have the issue on another function called by a Supabase client (with a valid JWT).

**Comment by GaryAustin1** [comment]
Service_role key should have access to any table in the public schema.  No other JWT is needed.
You don't show your tables, is it public or Public?  Capital letters matter.
Do you use Prisma?  Prisma tends to mess with the grant in the public schema to the point there is a doc guide on restoring them.

**Comment by ErwanJes** [comment]
Thanks for the answer!
It’s a pretty basic project.

Did I need to set something else?
I only have these instructions in the `serve()` function.

I also disabled the "Enforce JWT Verification".
I'd like to test it as a simple: ` curl -i --location --request POST 'http://localhost:54321/functions/v1/'`

The table name is lowercase and I don’t use Prisma.

Here is the full functions:
```
import { createClient } from '@supabase/supabase-js'

serve(async (req) => {
  
  const supabase = createClient(
    Deno.env.get("SUPABASE_URL") ?? "",
    Deno.env.get("SUPABASE_SERVICE_ROLE_KEY") ?? "",
  );
  
  const { data, error } = await supabase.from('profiles').select('*')
  console.log({ data, error })
  return new Response(
    JSON.stringify("{'message': 'OK'}"),
    { headers: { "Content-Type": "application/json" } },
  )
})
```

Here is a screenshot of a query in the project dashboard

<img width="385" alt="Capture d’écran 2023-07-17 à 08 44 16" src="https://github.com/supabase/supabase/assets/1706728/90f7368f-c759-4bab-9864-4243c0fdd081">

**Reply by ErwanJes** [reply]
To be honest, it feels like a stupid issue where the function is not linked with the project...
But I see it on my dashboard, I try to link it again and I don't see a difference when I'm running it locally

**Comment by ErwanJes** [comment]
Actually it works on the deployed version...
I don't know why. It's the exact same code
